### ISSUE TITLE
[BUG] "AttributeError: attr_data" when inserting hybrid entities on 0.4.0

### ISSUE THREAD

**Issue by tvial** [milvus-io/pymilvus#426]
**Describe the bug**
Using pymilvus 0.4.0 against Milvus server 0.10.6 CPU only.
I get an "AttributeError: attr_data" error upon calling `client.insert_hybrid()`

**Steps/Code to reproduce behavior**
Following the quick start tutorial on https://www.milvus.io/docs/v0.10.6/milvus_docker-cpu.md:

- Get & start milvusdb/milvus:0.10.6-cpu-d022221-64ddc2 docker container
- Install pymilvus 0.4.0

Then paste the following code (taken from #193 with minimal adaptations):

```
from milvus import Milvus, IndexType, MetricType, Status, DataType
import random

milvus = Milvus(host='localhost', port='19530')
collection_name = 'process'

collection_fields = [
    {"field_name": "A", "data_type": DataType.INT64},
    {"field_name": "B", "data_type": DataType.INT64},
    {"field_name": "C", "data_type": DataType.INT64},
    {"field_name": "Vec", "dimension": 128, "extra_params": {"index_file_size": 100, "metric_type": MetricType.L2}}
]
status = milvus.create_hybrid_collection(collection_name, collection_fields)

num = 10
dim = 128
A_list = [random.randint(0, 255) for _ in range(num)]
vec = [[random.random() for _ in range(dim)] for _ in range(num)]
hybrid_entities = [
    {"field_name": "A", "field_values": A_list},
    {"field_name": "B", "field_values": A_list},
    {"field_name": "C", "field_values": A_list},
]
vector_entities = [
    {"field_name": "Vec", "field_values": vec}
]

status, ids = milvus.insert_hybrid(collection_name, hybrid_entities, vector_entities)
```

**Expected behavior**
A status code of 0 and vectors inserted according to the schema provided.

Instead, the following stack trace is thrown:

```
---------------------------------------------------------------------------
AttributeError                            Traceback (most recent call last)
<ipython-input-11-5de750074521> in <module>
----> 1 status, ids = milvus.insert_hybrid(collection_name, hybrid_entities, vector_entities)

~/miniconda3/envs/xxx/lib/python3.7/site-packages/milvus/client/stub.py in inner(self, *args, **kwargs)
     35     @functools.wraps(func)
     36     def inner(self, *args, **kwargs):
---> 37         return func(self, *args, **kwargs)
     38 
     39     return inner

~/miniconda3/envs/xxx/lib/python3.7/site-packages/milvus/client/stub.py in insert_hybrid(self, collection_name, entities, vector_entities, ids, partition_tag, params)
    439     def insert_hybrid(self, collection_name, entities, vector_entities, ids=None, partition_tag=None, params=None):
    440         with self._connection() as handler:
--> 441             return handler.insert_hybrid(collection_name, entities, vector_entities, ids, partition_tag, params)
    442 
    443     def get_entity_by_id(self, collection_name, ids, timeout=None):

~/miniconda3/envs/xxx/lib/python3.7/site-packages/milvus/client/grpc_handler.py in handler(self, *args, **kwargs)
     53                 record_dict["Exception"] = str(datetime.datetime.now())
     54                 LOGGER.error("\nAddr [{}] {}\nExcepted error: {}\n\t{}".format(self.server_address, func.__name__, e, record_dict))
---> 55                 raise e
     56 
     57         return handler

~/miniconda3/envs/xxx/lib/python3.7/site-packages/milvus/client/grpc_handler.py in handler(self, *args, **kwargs)
     39                     self.ping()
     40                 record_dict["RPC start"] = str(datetime.datetime.now())
---> 41                 return func(self, *args, **kwargs)
     42             except grpc.FutureTimeoutError as e:
     43                 record_dict["RPC timeout"] = str(datetime.datetime.now())

~/miniconda3/envs/xxx/lib/python3.7/site-packages/milvus/client/grpc_handler.py in insert_hybrid(self, collection_name, entities, vector_entities, ids, tag, params)
    521     @error_handler([])
    522     def insert_hybrid(self, collection_name, entities, vector_entities, ids=None, tag=None, params=None):
--> 523         insert_param = Prepare.insert_hybrid_param(collection_name, tag, entities, vector_entities, ids, params)
    524         response = self._stub.InsertEntity(insert_param)
    525         status = response.status

~/miniconda3/envs/xxx/lib/python3.7/site-packages/milvus/client/prepare.py in insert_hybrid_param(cls, collection_name, tag, entities, vector_entities, ids, params)
    134             if isinstance(values, list):
    135                 if isinstance(values[0], int):
--> 136                     entity_param.attr_data.add(int_value=values)
    137                 elif isinstance(values[0], float):
    138                     entity_param.attr_data.add(double_value=values)

AttributeError: attr_data
```

**Environment details**

- Hardware/Softward conditionsUbuntu 20.04, CPU only, 8 GB RAM
- Method of installation: Docker (server) + pip install (pymilvus)
- Milvus version: 0.10.6 (server) + 0.4.0 (pymilvus)
- Milvus configuration: https://raw.githubusercontent.com/milvus-io/milvus/0.10.6/core/conf/demo/server_config.yaml

**Additional context**
Since #193 is closed and the hybrid APIs are exposed by the Milvus client, I expected the `insert_hybrid` call to succeed.
The hybrid-dev branch from https://github.com/BossZou/pymilvus (mentioned on #193) is no more available so I could not check if the missing attributes can be restored somehow.

**Comment by tvial** [comment]
If that's of any help, here is a quick patch that silences the error above and allows insertion of the entities. Now it's the `search_hybrid_pb` call that's failing with `AttributeError: int_value`, still in `prepare.py`.

I understand the hybrid API is simply not ready for release. Do you confirm, or should I use another combination of Milvus server + pymilvus versions?


[prepare.py.patch.txt](https://github.com/milvus-io/pymilvus/files/6051035/prepare.py.patch.txt)

**Comment by BossZou** [comment]
@tvial, this API is experimental, and the functionality has been moved to [milvus-experimental](https://github.com/milvus-io/milvus/tree/experimental). However, this branch will not be maintained. In the future, we'll release a new Milvus version (May be v2.x) to support DSL and distributed deployment.

**Comment by tvial** [comment]
OK thanks. I think you should make that very clear in the various pieces of documentation (website, milvus, pymilvus), and stick to a more readable version numbering scheme. For example I read somewhere (in another issue discussion) that pymilvus 0.4.0 is the latest while 0.5.x is deprecated, and you mention a milvus v2.x while there is no 1.x around.
Besides, the issue I reported applies to pymilvus, not milvus: those are functions in pymilvus that fail before even calling the server. Given that the milvus (server) master branch does have examples of hybrid querying (not only in the milvus-experimental branch you mention), you can imagine how confusing it can be for people like me who do not participate in the project.

All in all, sorry but I'm looking somewhere else for for our app, an open source engine like Milvus is a fantastic idea but I'm not confident enough to use it in a project.

**Comment by BossZou** [comment]
@tvial , I'm sorry to confuse you with the mismatched documentation. Actually we're preparing relevant documentation(website, milvus, pymilvus). 
We are going to release milvus v1.0.x which is based on milvus v0.10.x where hybrid operation is not supported, meanwhile to update relevant documentation to show the changes. And we're developing milvus v2.0.x which is of hybrid operation supported and cloud-native.
-----------------------------------------------------------------------------------------------------------------------------------------------------------------------
We were use to release a version named milvus v0.11.0 (corresponding to pymilvus v0.3.0) to support hybrid operation but we recalled it for some reason (you can refer [here](https://github.com/milvus-io/milvus/discussions/4793) to see more details). After that, we released a new pymilvus version v0.4.0 based on pymilvus v0.2.15 to make the latest pymilvus correspond to milvus v0.10.x. Therefore, the hybrid operation is Incomplete and experimental at pymilvus v0.4.0, but it is a bug the hybrid APIs is still maintained. We'll fix this.

**Comment by gujun720** [comment]
> OK thanks. I think you should make that very clear in the various pieces of documentation (website, milvus, pymilvus), and stick to a more readable version numbering scheme. For example I read somewhere (in another issue discussion) that pymilvus 0.4.0 is the latest while 0.5.x is deprecated, and you mention a milvus v2.x while there is no 1.x around.
> Besides, the issue I reported applies to pymilvus, not milvus: those are functions in pymilvus that fail before even calling the server. Given that the milvus (server) master branch does have examples of hybrid querying (not only in the milvus-experimental branch you mention), you can imagine how confusing it can be for people like me who do not participate in the project.
> 
> All in all, sorry but I'm looking somewhere else for for our app, an open source engine like Milvus is a fantastic idea but I'm not confident enough to use it in a project.

Hi @tvial ,

Sorry for the confusion. I have to admit you are right we have a really awful version numbering scheme. And because of the 0.11 drawback, there is some disconnection in the project documentation. 

We are working to remediate the disconnection and we will soon align the version scheme across different components in our first milestone release, which is Milvus V1.0.

I hope with the development of the Milvus project, we will get your confidence in the Milvus project.
